pkg/resources/manager/manager.go

```diff
@@ -326,6 +361,12 @@ func (r *resourceManager) deallocate(machine *api.Machine, deallocateResources c
 	return nil
 }
 
+func (r *resourceManager) deallocateUnassignResources(resources core.ResourceList) {
+	for _, s := range r.sources {
```

Comment:
Can't we use here the same mechanism that is being used above `s, ok := r.registredResources[key]` , in place of iterating over all the sources?

Reply:
fixed